File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {

        Map<Integer,Integer> map = new HashMap<>();

        for(int i: nums){
            if(map.containsKey(i)){
                return true;
            }
            map.put(i,0);
        }
        return false;
        
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:

        map={0:1}
        sm=0
        cnt=0
        for i in nums:
            sm+=i
            
            if sm-k in map:
                cnt+=map[sm-k]
            if sm not in map:
                map[sm]=1
            else:
                map[sm]+=1
        return cnt

        